### PR TITLE
mailspring: init at 1.7.8

### PR DESCRIPTION
**Pull request by toschmidt**
###### Motivation for this change

Packaged the mail client [Mailspring](https://github.com/Foundry376/Mailspring) at version 1.7.5

> Mailspring is a new version of Nylas Mail maintained by one of the original authors. It's faster, leaner, and shipping today!

Currently two pull requests for mailspring at version 1.6.3 exist: #69384 and #69027 

Syncing mails does not work in #69384 and #69027 uses buildFHSUserEnv although only one file needs patching.

This PR is based on the work done by rummik in #69384 and solves both problems for the latest version of mailspring.

The application needs `gnome-keyring` for storing passwords.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers
cc @

### CODE REVIEW

**Comment by drewrisinger** [comment]
Closes #69027 #69384
@GrahamcOfBorg build mailspring

File: pkgs/applications/networking/mailreaders/mailspring/default.nix
Comment (drewrisinger) [inline pkgs/applications/networking/mailreaders/mailspring/default.nix:53]: I'm honestly not too familiar with policies on using dpkg (i.e. pre-compiled binaries) in Nix. I know general preference is to build from source if at all possible, but it seems this ISN'T possible due to a private C++ section of ``mailspring``? Is that correct? Did you try building from the GitHub at all?

File: pkgs/applications/networking/mailreaders/mailspring/default.nix
Comment (drewrisinger) [inline pkgs/applications/networking/mailreaders/mailspring/default.nix:15]: Sort these alphabetically

File: pkgs/applications/networking/mailreaders/mailspring/default.nix
Comment (drewrisinger) [inline pkgs/applications/networking/mailreaders/mailspring/default.nix:44]: Sort these & other package lists below alphabetically

File: pkgs/applications/networking/mailreaders/mailspring/default.nix
Comment (drewrisinger) [inline pkgs/applications/networking/mailreaders/mailspring/default.nix:63]:
```suggestion
      --replace realpath ${coreutils}/bin/realpath \
      --replace dirname ${coreutils}/bin/dirname
```

File: pkgs/applications/networking/mailreaders/mailspring/default.nix
Comment (drewrisinger) [inline pkgs/applications/networking/mailreaders/mailspring/default.nix:66]:
```suggestion
    ln -s ${openssl}/lib/libcrypto.so $out/lib/libcrypto.so.1.0.0
```

File: pkgs/applications/networking/mailreaders/mailspring/default.nix
Comment (drewrisinger) [inline pkgs/applications/networking/mailreaders/mailspring/default.nix:83]:
```suggestion
    platforms = platforms.x86_64;
```

File: pkgs/applications/networking/mailreaders/mailspring/default.nix
Comment (drewrisinger) [inline pkgs/applications/networking/mailreaders/mailspring/default.nix:82]:
```suggestion
    homepage = "https://getmailspring.com";
    downloadPage = "https://github.com/Foundry376/Mailspring";
```
Quote per RFC 39? RFC49?

File: pkgs/top-level/all-packages.nix
Comment (drewrisinger) [inline pkgs/top-level/all-packages.nix:26447]: This should be inserted alphabetically-ish. I would probably place it near thunderbird, there might be other mail clients in there? ``all-packages`` is loosely sorted by category then name.

**Review by drewrisinger — CHANGES_REQUESTED** [review]
Thanks for submitting your first PR! This looks like a well-researched PR. I'm honestly not terribly familiar with the Linux binary side of nixpkgs, but I know the general preference is to build packages from scratch in Nixpkgs if possible.

There's some formatting/best practices included below.
Also, when trying to review this, I ran into a merge conflict with the maintainers list (probably b/c you submitted a different PR & now there's a conflict). So please remove that commit from this with ``git rebase -i master``. Reviewers can't easily test building this until that's fixed.

File: pkgs/applications/networking/mailreaders/mailspring/default.nix
Comment (toschmidt) [reply pkgs/applications/networking/mailreaders/mailspring/default.nix:52]: @drewrisinger thanks for your review.

Yes, mailspring depends on a private repo, that implements syncing mails. I found no other way to get the binary than depackaging one of the provided builds.

File: pkgs/applications/networking/mailreaders/mailspring/default.nix
Comment (toschmidt) [reply pkgs/applications/networking/mailreaders/mailspring/default.nix:15]: Ok

File: pkgs/applications/networking/mailreaders/mailspring/default.nix
Comment (toschmidt) [reply pkgs/applications/networking/mailreaders/mailspring/default.nix:43]: Ok

File: pkgs/top-level/all-packages.nix
Comment (toschmidt) [reply pkgs/top-level/all-packages.nix:26447]: I placed it near thunderbird and mutt.

File: pkgs/applications/networking/mailreaders/mailspring/default.nix
Comment (toschmidt) [reply pkgs/applications/networking/mailreaders/mailspring/default.nix:64]: without `${openssl.out}` build fails

**Comment by toschmidt** [comment]
@drewrisinger  thanks for your review.

> Also, when trying to review this, I ran into a merge conflict with the maintainers list (probably b/c you submitted a different PR & now there's a conflict). So please remove that commit from this with `git rebase -i master`. Reviewers can't easily test building this until that's fixed.

I removed myself from the maintainer list (it's my only PR so there should not be any conflict in that file). Before merging, we should probably revert that unless someone else wants to maintain the package. I also updated mailspring to the latest version (1.7.6) and applied your suggestions.

**Comment by drewrisinger** [comment]
Oh, sorry. If this is actually your first package, then a commit adding yourself as maintainer IS appropriate. Just make sure that it merges with ``master`` branch. I recommend rebasing, dropping the revert commit, and then ``git push -f`` that fixed branch. You should resolve whatever merge commit the maintainer list pops up.

Builds locally with ``nixpkgs-review pr 85492``
```
https://github.com/NixOS/nixpkgs/pull/85492
1 package built:
mailspring
```

**Comment by drewrisinger** [comment]
Note: commits will also need squashed down to just the maintainer commit & ``mailspring: init at 1.7.6`` (PR title should also be changed) before this PR is accepted/merged.

**Comment by toschmidt** [comment]
Ok, the PR should have no conflicts with master and I squashed all commits down.

File: pkgs/applications/networking/mailreaders/mailspring/default.nix
Comment (drewrisinger) [inline pkgs/applications/networking/mailreaders/mailspring/default.nix:67]: why the comment here?

**Review by drewrisinger — COMMENTED** [review]
Few small notes still to be resolved by someone who knows more about packaging ``.deb`` better than I.

Builds locally, not tested.
``nixpkgs-review pr 85492``
```
https://github.com/NixOS/nixpkgs/pull/85492
1 package built:
mailspring
```

**Comment by drewrisinger** [comment]
Closes #69027

**Comment by drewrisinger** [comment]
Closes #69384

**Comment by tyrion** [comment]
I tried upgrading this to the latest Mailspring release and it seems to work fine so far.
You just need to change the version and sha checksum:

```
  version = "1.7.8";

  src = fetchurl {
    url = "https://github.com/Foundry376/Mailspring/releases/download/${version}/mailspring-${version}-amd64.deb";
    sha256 = "207fbf813b6da018a5b848e5dc1194b5996daab39adbd873b2cecb0565c105ce";
  };
```

File: pkgs/applications/networking/mailreaders/mailspring/default.nix
Comment (tyrion) [inline pkgs/applications/networking/mailreaders/mailspring/default.nix:78]: Mailspring is unfortunately not free software. This should be changed to "unfree".

**Review by tyrion — CHANGES_REQUESTED** [review]

**Review by drewrisinger — APPROVED** [review]
* [x] Diff LGTM
* [x] Builds Locally
* [x] After build, GUI launches
* [x] Still not sure about best method when using dpkg, but seems to build fine.

**Review by tobkratz — APPROVED** [review]
I can get it work. 
LGTM

**Review by danieldk — APPROVED** [review]
Thanks for submitting this and sorry for having to wait so long for getting this merged. It seems that mailspring relies on wget for crash dumps:

```
$ wrapit --bind /tmp/.X11-unix/X0 /tmp/.X11-unix/X0 --dir /home/daniel --setenv DISPLAY :0 `realpath results/mailspring/bin/mailspring`
Fontconfig error: Cannot load default config file
Running database migrations
App load time: 145ms

Running Setup
{"error":null}
Cannot upload crash dump: cannot exec /usr/bin/wget
Failed to get crash dump id.
Report Id: 
Cannot upload crash dump: cannot exec /usr/bin/wget
Failed to get crash dump id.
Report Id: 
```

Could be something to fix for a future PR.

**Pull request merged**
